[CI] TopNOperatorTests testSplitOnSize failing

**Build Scans:**
- [elasticsearch-intake #39292 / part3](https://gradle-enterprise.elastic.co/s/ygibidgf57ujq)
- [elasticsearch-pull-request #143985 / part-3](https://gradle-enterprise.elastic.co/s/bdaitaxolc7nw)
- [elasticsearch-pull-request #143985 / part-3](https://gradle-enterprise.elastic.co/s/bxlbwwqklbg7o)
- [elasticsearch-pull-request #143956 / part-3](https://gradle-enterprise.elastic.co/s/gvsuqwffmch6g)
- [elasticsearch-pull-request #143956 / part-3](https://gradle-enterprise.elastic.co/s/opv3dqd22xw24)
- [elasticsearch-pull-request #143504 / part-3](https://gradle-enterprise.elastic.co/s/ftychavth7ggq)
- [elasticsearch-pull-request #143504 / part-3](https://gradle-enterprise.elastic.co/s/6s5qxgalvg6je)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:compute:test" --tests "org.elasticsearch.compute.operator.topn.TopNOperatorTests.testSplitOnSize" -Dtests.seed=636025157CDDA0D1 -Dtests.locale=mi -Dtests.timezone=America/Thule -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.compute.operator.topn.TopNOperatorTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSplitOnSize'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: (a value equal to or greater than <16384L> and a value less than or equal to <35406L>)
     but: a value equal to or greater than <16384L> <1712L> was less than <16384L>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testSplitOnSize
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 7 failures in test testSplitOnSize (1.7% fail rate in 424 executions)
- [main] 6 failures in step part-3 (2.0% fail rate in 295 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (1.0% fail rate in 286 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -271,9 +271,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.ForecastIT
   method: testOverflowToDisk
   issue: https://github.com/elastic/elasticsearch/issues/148606
-- class: org.elasticsearch.compute.operator.topn.TopNOperatorTests
-  method: testSplitOnSize
-  issue: https://github.com/elastic/elasticsearch/issues/148613
 - class: org.elasticsearch.smoketest.SmokeTestWatcherTestSuiteIT
   method: testMonitorClusterHealth
   issue: https://github.com/elastic/elasticsearch/issues/148615

--- a/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/topn/TopNOperatorTests.java
+++ b/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/topn/TopNOperatorTests.java
@@ -2200,10 +2200,13 @@ public class TopNOperatorTests extends OperatorTestCase {
                 try (Page out = op.getOutput()) {
                     totalPositions += out.getPositionCount();
                     if (totalPositions < expectedTotal) {
-                        assertThat(out.ramBytesUsedByBlocks(), both(greaterThanOrEqualTo(minPageSize)).and(lessThanOrEqualTo(maxPageSize)));
-                    } else {
-                        assertThat(out.ramBytesUsedByBlocks(), lessThanOrEqualTo(maxPageSize));
+                        // We can over-allocate to one page for BytesRefArray that requires less than one page.
+                        // So don't check the min page size if jumboPageBytes is less than a page.
+                        if (minPageSize >= PageCacheRecycler.PAGE_SIZE_IN_BYTES) {
+                            assertThat(out.ramBytesUsedByBlocks(), greaterThanOrEqualTo(minPageSize));
+                        }
                     }
+                    assertThat(out.ramBytesUsedByBlocks(), lessThanOrEqualTo(maxPageSize));
                 }
             }
             assertThat(totalPositions, equalTo(expectedTotal));

